docs/pages/deploy-a-cluster/high-availability.mdx

```diff
@@ -0,0 +1,520 @@
+`/etc/teleport.yaml`. We will explain the required configuration fields for a
+high-availability Teleport deployment below. These are the minimum requirements,
+and when planning your high-availability deployment, you will want to follow a
+more specific [deployment guide](introduction.mdx) for your environment. 
+
+#### `storage` 
+
+The first configuration section to write is the `storage` section, which
+configures the key-value store and object store for the Teleport Auth Service.
+
+The example configurations below apply to Teleport deployments that use Amazon
+DynamoDB, Google Cloud Firestore, and etcd as the Auth Service backend:
+
+<Tabs>
+<TabItem label="Amazon DynamoDB">
+
+```yaml
+version: v2
+teleport:
+  storage:
+    type: dynamodb
+    region: "us-east-1" # Or another region
+    table_name: "my-kv-table"
+    audit_events_uri: ["dynamodb://my-audit-events-table", "stdout://"]
+    continuous_backups: true
+    audit_sessions_uri: "s3://my-session-recording-bucket"
+```
+</TabItem>
+<TabItem label="Google cloud Firestore">
+```yaml
+version: v2
+teleport:
+  storage:
+    type: firestore
+    project_id: my-google-cloud-project
+    collection_name: my-cluster-state-table
+    credentials_path: /etc/teleport-secrets/gcp-credentials.json
+    audit_events_uri: ['firestore://my-audit-events-table?projectID=my-google-cloud-project&credentialsPath=/etc/teleport-secrets/gcp-credentials.json', 'stdout://']
+    audit_sessions_uri: "gs://session-recording-bucket?projectID=my-google-cloud-project&credentialsPath=/etc/teleport-secrets/gcp-credentials.json"
+```
+</TabItem>
+<TabItem label="etcd">
+```yaml
+version: v2
+teleport:
+  storage:
+    type: etcd
+    peers: ["https://172.17.0.1:4001", "https://172.17.0.2:4001"]
+    tls_cert_file: /var/lib/teleport/etcd-cert.pem
+    tls_key_file: /var/lib/teleport/etcd-key.pem
+    tls_ca_file: /var/lib/teleport/etcd-ca.pem
+    username: username
+    password_file: /mnt/secrets/etcd-pass
+    prefix: /teleport/
+    insecure: false
+    etcd_max_client_msg_size_bytes: 15728640
+    audit_sessions_uri: "s3://my-session-recording-bucket"
+```
+</TabItem>
+</Tabs>
+
+In the `storage` section, the `audit_sessions_uri` field names the bucket within
+Google Cloud Storage or an S3-compatible service to use for session recordings,
+and the remaining fields are specific to each backend. 
+
+Consult our [Backends Reference](../reference/backends.mdx) for the values you
+should use.  
+
+#### `auth_service` and `proxy_service` 
+
+The `auth_service` and `proxy_service` sections configure the Auth Service and
+Proxy Service, which we will run together on each Teleport instance. The
+configuration will depend on whether you are enabling TLS Routing in your
+cluster:
+
+<Tabs>
+<TabItem label="TLS Routing">
+
+To enable TLS Routing in your Teleport cluster, add the following to your
+Teleport configuration:
+
+```yaml
+version: v2
```

Comment:
`v3` is the latest config version, and to my knowledge the docs were all updated to use v3 where possible. Can we stick with v3 here as well?